src/algorithms/all_paths.c

```diff
@@ -25,52 +26,55 @@ void _AllPaths
     Path **paths            // Paths constructed.
 )
 {
-        if(hop >= minHops && hop <= maxHops) {
-            Path clone = Path_clone(*path);
+    if(hop >= minHops && hop <= maxHops) {
+        Path clone = Path_clone(*path);
 
-            if(*pathsCount >= *pathsCap) {
-                (*pathsCap) = (*pathsCap) * 2 + 1;
-                (*paths) = realloc(*paths, sizeof(Path) * (*pathsCap));
-            }
-
-            (*paths)[*pathsCount] = clone;
-            (*pathsCount) = (*pathsCount) + 1;
+        if(*pathsCount >= *pathsCap) {
+            (*pathsCap) = (*pathsCap) * 2 + 1;
```

Comment:
Why + 1? The initial cap is a power of 2.

Reply:
Can't recall why, changed to `(*pathsCap) = (*pathsCap) * 2;`